components/suggest/src/rs.rs

```diff
@@ -100,30 +100,44 @@ pub(crate) enum SuggestRecord {
     GlobalConfig(DownloadedGlobalConfig),
 }
 
+/// Implemented by the various attachment types
+pub trait SuggestAttachment: DeserializeOwned + std::fmt::Debug {
+    /// Get fields that we saw when ingesting this attachment, but don't recognize
+    fn unknown_fields(&self) -> &serde_json::Map<String, serde_json::Value>;
+
+    fn has_unknown_fields(&self) -> bool {
+        !self.unknown_fields().is_empty()
+    }
+}
+
 /// Represents either a single value, or a list of values. This is used to
 /// deserialize downloaded attachments.
 #[derive(Clone, Debug, Deserialize)]
 #[serde(untagged)]
-enum OneOrMany<T> {
+pub enum OneOrMany<T> {
     One(T),
     Many(Vec<T>),
 }
 
-/// A downloaded Remote Settings attachment that contains suggestions.
-#[derive(Clone, Debug, Deserialize)]
-#[serde(transparent)]
-pub(crate) struct SuggestAttachment<T>(OneOrMany<T>);
-
-impl<T> SuggestAttachment<T> {
+impl<T> OneOrMany<T> {
     /// Returns a slice of suggestions to ingest from the downloaded attachment.
```

Comment:
Unrelated to this patch, but this comment needs some updates as the attachment payload could be either suggestions or provider configurations now.

Reply:
I ended up adding the note about what items an attachment payload can have top-level docstring and removing this docstring altogether.  I think it's more like to remain correct going forward that way.